### ISSUE TITLE
Package for ROS Melodic

### ISSUE THREAD

**Issue by mbusy** [ros-naoqi/libqi-release#6]
This issue is basically the same as #5, for the Melodic distro. Having naoqi_libqi (and naoqi_libqicore) released for Melodic would really be useful to researchers

**Comment by mbusy** [comment]
Small update on the subject: I adapted the code of libqi-release ([here](https://github.com/Pandhariix/libqi-release/tree/release/melodic/naoqi_libqi)) and libqicore-release ([here](https://github.com/Pandhariix/libqicore-release/tree/release/melodic/naoqi_libqicore)) so that they could be compiled with melodic.

I am not completely familiar with the package release process, but from what I understand looking at the network of the repo, the modifications could be committed to "release/melodic/naoqi_libqi(core)" branches, before releasing the packages for melodic. I would like to submit PRs, but such branches don't yet exist on the base repositories.

@suryaambrose, @mikaelarguedas, how should I proceed ?

**Comment by mikaelarguedas** [comment]
As a general rule, code change should always live in the source repositories and **not** be pushed by hand to the `-release` repositories.
Release repositories take a snapshot of the source repositories (import a specific version) and then apply release patches + automagically add the files required by the packaging system of the targeted platforms (e.g. Debian control and rule files).

Answers needed before releasing:
- What version of the software (libqi / libqi_core) should be released in Melodic?
- Does that version have a release branch / tag on the source repository?
  - does it build as is or requires additional patches?

Then the release process is highlighted in http://wiki.ros.org/bloom/Tutorials/ReleaseThirdParty

Basically you need to:
- push a valid catkin package.xml (listing all the dependencies of the package) to this repo master branch under a `melodic` folder
- run bloom
  - create a track for Melodic matching the other releases (use :{ask} for version and release tag and use `melodic` as the patch folder)
  - open the automatically generated PR to ros/rosdistro

HTH,

Edit: some info at https://github.com/ros-naoqi/libqi-release/pull/2#issuecomment-188300455 too

**Comment by mbusy** [comment]
Thanks, yes that helps :)

> What version of the software (libqi / libqi_core) should be released in Melodic?

I would say 2.5.0/2.3.1, the ones used for the kinetic release (I did my tests with these versions), but that's debatable

> Does that version have a release branch / tag on the source repository? does it build as is or requires additional patches?

If we select the versions used for the kinetic release, then yes. Yep, additional patches are required in order to build

One last question:
I don't have the permission to edit this repo. Adding a package.xml in a melodic folder shouldn't be a problem (I will submit a PR), but for the second step (running bloom) my guess is that you need to have access to the base repo right ?

**Comment by mikaelarguedas** [comment]
> I would say 2.5.0/2.3.1the ones used for the kinetic release (I did my tests with these versions), but that's debatable

As this repo points to 2.7.0 for lunar, it seems to me that it's make more sense to target a more recent version and minimise dependency descrepancies (like boost). Not sure what OS / dep versions 2.9.0 supports but it seems like a good/recent candidate.
Then if it makes more sense to focus on 2.5.0 for some reason I'd be fine with that as well but will require more patching which in general I prefer to avoid (whatever can be fixed / is already fixed upstream should be reused)

> I don't have the permission to edit this repo. 

As the main maintainer/releaser of these packages, my conviction is that you should have access to all repos of this organization.
@nlyubova @zygopter @suryaambrose are you ok with me promoting Maxime as an owner of `ros-naoqi`?

> (I will submit a PR), but for the second step (running bloom) my guess is that you need to have access to the base repo right ?

I apparently still have power here so I can merge / run bloom if needed, but hopefully you'll unlock god-mode before then ;)

**Comment by nlyubova** [comment]
@mikaelarguedas fine for me

**Comment by mikaelarguedas** [comment]
@Pandhariix congrats you're now an owner of `ros-naoqi` !

Noticed 2FA is disabled on your account, I'd recommend enabling it to avoid bad surprises :)

**Comment by mbusy** [comment]
I thought the repo was pointing towards 2.5.0 for lunar, my bad... In any case you're right it would make more sense to target a recent version, I'll try to see how the 2.9.0 performs :)

Thanks for the god-mode! (2FA now enabled)

**Comment by mikaelarguedas** [comment]
> I thought the repo was pointing towards 2.5.0 for lunar, my bad...

Actually you're right it's pointing to 2.5.0, not sure where I got the idea of 2.7 from.
I guess I mixed up 2.5.7 and 2.7 when rereading the [lunar thread](https://github.com/ros-naoqi/libqi-release/issues/5#issuecomment-347187764) >_<.

**Comment by mbusy** [comment]
Another small update: I managed to compile naoqi_libqi and naoqi_libqicore for melodic, using the 2.9.0 releases of libqi and libqicore (patches to be applied on the upstream code [here](https://gist.github.com/Pandhariix/47f1bdf64ba58fb2a5d90ca6c05d6a2a)).

A few adaptations are required in naoqi_driver to be able to compile it (melodic_compatibility branch of [this fork](https://github.com/Pandhariix/naoqi_driver/tree/melodic_compatibility)), but the code is still compatible with older ros distros.

The driver seems to be running fine, I'll eventually release the packages for melodic next week :)

**Comment by iProzd** [comment]
> Small update on the subject: I adapted the code of libqi-release ([here](https://github.com/Pandhariix/libqi-release/tree/release/melodic/naoqi_libqi)) and libqicore-release ([here](https://github.com/Pandhariix/libqicore-release/tree/release/melodic/naoqi_libqicore)) so that they could be compiled with melodic.
> 
> I am not completely familiar with the package release process, but from what I understand looking at the network of the repo, the modifications could be committed to "release/melodic/naoqi_libqi(core)" branches, before releasing the packages for melodic. I would like to submit PRs, but such branches don't yet exist on the base repositories.
> 
> @suryaambrose, @mikaelarguedas, how should I proceed ?

Thanks for your update, but the links seem to be lost, have you got any new accessible update?

**Comment by mbusy** [comment]
Extra modifications were added to the repos, the snippets weren't relevant anymore. You can check the release/melodic/naoqi_libqi and release/melodic/naoqi_libqicore branches of the repos if you want to compile naoqi_libqi and naoqi_libqicore by yourself.

Or you could wait for the pacakges to be released (hopefully pretty soon)

**Comment by mbusy** [comment]
Bit of a necropost, but the package was released for melodic along with other useful ROS packages for Pepper and NAO (see [Melodic sync 2020-01-16](https://discourse.ros.org/t/new-packages-for-melodic-2020-01-16/12356))
